mautrix_signal/commands/signal.py

```diff
@@ -287,3 +288,77 @@ async def raw(evt: CommandEvent) -> None:
             await evt.reply(
                 f"Got reply `{resp_type}`:\n\n```json\n{json.dumps(resp_data, indent=2)}\n```"
             )
+
+
+@command_handler(
+    needs_auth=True,
+    management_only=False,
+    help_section=SECTION_SIGNAL,
+    help_text=("Create a Signal group for the current Matrix room. "),
+)
+async def create(evt: CommandEvent) -> EventID:
+    if await po.Portal.get_by_mxid(evt.room_id):
+        return await evt.reply("This is already a portal room.")
+
+    title, about, levels, encrypted, avatar_url = await get_initial_state(
+        evt.az.intent, evt.room_id
+    )
+    #    if not title:
+    #        return await evt.reply("Please set a title before creating a Telegram chat.")
+
+    portal = po.Portal(
+        chat_id=GroupID(""),
+        mxid=evt.room_id,
+        name=title,
+        topic=about,
+        encrypted=encrypted,
+        receiver="",
+        avatar_url=avatar_url,
+    )
+
+    await warn_missing_power(levels, evt)
+
+    try:
+        await portal.create_signal_group(evt.sender, levels)
+    except ValueError as e:
+        await portal.delete()
+        return await evt.reply(e.args[0])
+    return await evt.reply(f"Signal chat created. ID: {portal.chat_id}")
+
+
+async def get_initial_state(
+    intent: IntentAPI, room_id: RoomID
+) -> tuple[str | None, str | None, PowerLevelStateEventContent | None, bool, ContentURI | None]:
+    state = await intent.get_state(room_id)
+    title: str | None = None
+    about: str | None = None
+    levels: PowerLevelStateEventContent | None = None
+    encrypted: bool = False
+    avatar_url: ContentURI | None = None
+    for event in state:
+        try:
+            if event.type == EventType.ROOM_NAME:
+                title = event.content.name
+            elif event.type == EventType.ROOM_TOPIC:
+                about = event.content.topic
+            elif event.type == EventType.ROOM_POWER_LEVELS:
+                levels = event.content
+            elif event.type == EventType.ROOM_CANONICAL_ALIAS:
+                title = title or event.content.canonical_alias
+            elif event.type == EventType.ROOM_ENCRYPTION:
+                encrypted = True
+            elif event.type == EventType.ROOM_AVATAR:
+                avatar_url = event.content.url
+        except KeyError:
+            # Some state event probably has empty content
+            pass
+    return title, about, levels, encrypted, avatar_url
+
+
+async def warn_missing_power(levels: PowerLevelStateEventContent, evt: CommandEvent) -> None:
+    if levels.get_user_level(evt.az.bot_mxid) < levels.redact:
+        await evt.reply(
+            "Warning: The bot does not have privileges to redact messages on Matrix. "
+            "Message deletions from Signal will not be bridged unless you give "
+            f"redaction permissions to [{evt.az.bot_mxid}](https://matrix.to/#/{evt.az.bot_mxid})"
```

Comment:
This error isn't applicable here since Signal only allows the sender to delete messages (and generally doesn't try to hide who deleted a message). Checking for invite and/or state default levels with a better message might still make sense though